### PR TITLE
roachtest: run EXPLAIN ANALYZE when tpchvec/perf fails

### PR DESCRIPTION
**Pull request by yuzefovich**
**workload: switch TPCH queries mapping from using string to int**

It also exports `NumQueries` and `QueriesByNumber` fields of `tpch`
package to be used by `tpchvec` roachtest.

Release note: None

**roachtest: minor cleanup of tpchbench**

This commit refactors `tpchbench` test slightly in order to hide `tpch`
bench type from the whole `roachtest` package.

Release note: None

**roachtest: run EXPLAIN ANALYZE when tpchvec/perf fails**

We have a mysterious rare failure on query 7 in which `vectorize=on`
performs significantly worse than `vectorize=off`. I'm out of possible
explanations for this, so this commit adds an ability to run `EXPLAIN
ANALYZE` on the query with both `vectorize` options when the slowness
threshold is exceeded. Hopefully it'll give us some insight into the
perf failures.

Addresses: #47401.

Release note: None

### CODE REVIEW

**Review by asubiotto — COMMENTED** [review]
:lgtm:

> Reviewed 3 of 3 files at r1, 2 of 2 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49397#-:-M7vaqpwBEijraOuAObE:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)

___
*[pkg/cmd/roachtest/tpchvec.go, line 56 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/49397#-M7v_MmW-zXzo4pu-YKD:-M7v_MmX-VpuUzn9FtJ0:b-qjja7z) ([raw file](https://github.com/cockroachdb/cockroach/blob/eb248de97fad8677357000086428b077d15a92d3/pkg/cmd/roachtest/tpchvec.go#L56)):*
> ```Go
> }
> 
> func vectorizeOptionToSetting(option bool, version crdbVersion) string {
> ```

nit: I think it'd be nice to have a more descriptive parameter (e.g. just `vectorize`)
___
*[pkg/cmd/roachtest/tpchvec.go, line 281 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/49397#-M7vafnF9ua7oKUFXwhr:-M7vafnF9ua7oKUFXwhs:biy8zfv) ([raw file](https://github.com/cockroachdb/cockroach/blob/eb248de97fad8677357000086428b077d15a92d3/pkg/cmd/roachtest/tpchvec.go#L281)):*
> ```Go
> 						t.Fatal(err)
> 					}
> 					t.Status(fmt.Sprintf("EXPLAIN ANALYZE with vectorize=%s url:\n%s", vectorizeSetting, url))
> ```

It might be a bit hard to parse this message due to multiple runs of the same query. What I would do is try to resuse the parsing code that gets the time and only print out the EXPLAIN ANALYZE url with the slowest time.
___
*[pkg/workload/tpch/expected_rows.go, line 20 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49397#-M7vYgjb1LvAmQsZ_-cF:-M7vYgjb1LvAmQsZ_-cG:b-yos3xr) ([raw file](https://github.com/cockroachdb/cockroach/blob/c17974b88e0b0057ba8618e7683b76d2344507f3/pkg/workload/tpch/expected_rows.go#L20)):*
> ```Go
> 		16: 18314,
> 	}
> 	queriesToCheckOnlyNumRows = map[int]bool{
> ```

nit: do we need this map? It should be sufficient to check for existence in `numExpectedRowsByQueryNumber`
___
*[pkg/workload/tpch/queries.go, line 14 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49397#-M7vYrVw5NgM5XSGuINR:-M7vYrVxC8zBg16oUX1J:b-7njrwc) ([raw file](https://github.com/cockroachdb/cockroach/blob/c17974b88e0b0057ba8618e7683b76d2344507f3/pkg/workload/tpch/queries.go#L14)):*
> ```Go
> 
> // NumQueries specifies the number of queries in TPC-H benchmark.
> const NumQueries = 22
> ```

nit: maybe make this a var and `NumQueries = len(QueriesByNumber)`

**Review by yuzefovich — COMMENTED** [review]
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49397#-:-M7wtMYXCHEXeoSMH7Nv:b-oduykh)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @asubiotto)

___
*[pkg/cmd/roachtest/tpchvec.go, line 281 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/49397#-M7vafnF9ua7oKUFXwhr:-M7wl_lQ-qwMTG29-B0-:bg3rn9w) ([raw file](https://github.com/cockroachdb/cockroach/blob/eb248de97fad8677357000086428b077d15a92d3/pkg/cmd/roachtest/tpchvec.go#L281)):*
<details><summary><i>Previously, asubiotto (Alfonso Subiotto Marqués) wrote…</i></summary><blockquote>

It might be a bit hard to parse this message due to multiple runs of the same query. What I would do is try to resuse the parsing code that gets the time and only print out the EXPLAIN ANALYZE url with the slowest time.
</blockquote></details>

Hm, I don't think parsing will be that hard.

Note that this `EXPLAIN ANALYZE` will run only on the query that we detected to be slow, 3 times with `on` and 3 times with `off`, so we will have 6 explain urls followed by message that "vec ON is slower ...". I don't think there is any parsing code that be reused - existing parsing code works on the output of `tpch` workload whereas here we will be getting the output from CLI.
___
*[pkg/workload/tpch/expected_rows.go, line 20 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/49397#-M7vYgjb1LvAmQsZ_-cF:-M7wnI5IAVKicROZrgW9:bfbq99i) ([raw file](https://github.com/cockroachdb/cockroach/blob/c17974b88e0b0057ba8618e7683b76d2344507f3/pkg/workload/tpch/expected_rows.go#L20)):*
<details><summary><i>Previously, asubiotto (Alfonso Subiotto Marqués) wrote…</i></summary><blockquote>

nit: do we need this map? It should be sufficient to check for existence in `numExpectedRowsByQueryNumber`
</blockquote></details>

Good point, done.
